tasks/deploy/coordinators/ezeth.ts

```diff
@@ -0,0 +1,132 @@
+import { task, types } from "hardhat/config";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+dayjs.extend(duration);
+import { parseEther, toFunctionSelector, zeroAddress } from "viem";
+import { zAddress } from "../utils";
+import { z } from "zod";
+import { DeployCoordinatorsBaseParams } from "./shared";
+
+export let zEzETHCoordinatorDeployConfig = z.object({
+  ezeth: zAddress.optional(),
+});
+
+export type EzETHCoordinatorDeployConfigInput = z.input<
+  typeof zEzETHCoordinatorDeployConfig
+>;
+
+export type EzETHCoordinatorDeployConfig = z.infer<
+  typeof zEzETHCoordinatorDeployConfig
+>;
+
+export type DeployCoordinatorsEzethParams = {
+  admin?: string;
+};
+
+task("deploy:coordinators:ezeth", "deploys the EzETHdeployment coordinator")
+  .addOptionalParam("admin", "admin address", undefined, types.string)
+  .setAction(
+    async (
+      { admin }: DeployCoordinatorsEzethParams,
+      {
+        deployments,
+        run,
+        network,
+        viem,
+        getNamedAccounts,
+        config: hardhatConfig,
+      },
+    ) => {
+      // Retrieve the HyperdriveFactory deployment artifact for the current network
+      let factory = await deployments.get("HyperdriveFactory");
+      let factoryAddress = factory.address as `0x${string}`;
+
+      // Set the admin address to the deployer address if one was not provided
+      if (!admin?.length) admin = (await getNamedAccounts())["deployer"];
+
+      // Deploy a mock Lido contract if no adress was provided
+      let ezeth = hardhatConfig.networks[network.name].coordinators?.ezeth;
+      if (!ezeth?.length) {
+        let mockEzEth = await viem.deployContract("MockEzEthPool", [
+          parseEther("0.035"),
+          admin as `0x${string}`,
+          true,
+          parseEther("500"),
+        ]);
+        await deployments.save("MockEzEthPool", {
+          ...mockEzEth,
+          args: [
+            parseEther("0.035"),
+            admin as `0x${string}`,
+            true,
+            parseEther("500"),
+          ],
+        });
```

Comment:
It would be nice to be able to parameterize deployments of vaults. Can totally leave for later, but my guess is that we could just add another Zod schema and add it as an extension to the hardhat configuration.

Reply:
Totally agree, it's pretty tough to track all the inline definitions. Having it in the config with everything else makes a lot of sense to me